src/inscription.rs

```diff
@@ -0,0 +1,377 @@
+use {
+  crate::{Ordinal, Transaction, VecDeque},
```

Comment:
Figure out how we can use `use super::*` here without conflicting Error types. Maybe just rename `Error` type here? What do we do in other modules with custom error types?